app/src/main/java/edu/uc/forbesne/shoppingsidekick/dto/CartItem.kt

```diff
@@ -10,8 +10,9 @@ data class CartItem (var UPC : String ="", var quantity: Int = 0, var imageURL :
     }
 
     fun reduce(quantity: Int){
-        this.quantity -= quantity
-        if(this.quantity < 0) this.quantity = 0
+        if(this.quantity > 0) {
```

Comment:
Instead of checking the quantity after the change (and then reverting the change), check the quantity before making changes.

Reply:
This appears to be different from what they were doing.  They were setting a floor of 0 on quantity.  If quantity were (theoretically) less than 0 coming in, it would still be less than 0 with this new approach.

In any case, the old approach is fine, so I'd keep it.  I don't see any technical debt reduction with the new approach.